src/lib.rs

```diff
@@ -0,0 +1,450 @@
+extern crate sxd_document;
+
+use std::str;
+use std::string::String;
+use std::collections::HashMap;
+
+use sxd_document::{
+    parser,
+    Package,
+};
+use sxd_document::dom::{
+    Document,
+    Root,
+    ChildOfRoot,
+    Element,
+    ChildOfElement,
+};
+
+#[derive(Debug)]
+pub struct Program {
+    pub groups: Vec<StatementBody>
+}
+
+#[derive(PartialEq, Debug)]
+pub struct StatementBody {
+    pub blocks: Vec<Block>
+}
+
+#[derive(PartialEq, Debug)]
+pub struct Block {
+    pub block_type: String,
+    pub id: String,
+    pub fields: Option<HashMap<String, FieldValue>>,
+    pub statements: Option<HashMap<String, StatementBody>>,
+}
+
+#[derive(PartialEq, Debug)]
+pub enum FieldValue {
+    SimpleField(String),
+    ExpressionField(Block),
+}
+
+impl Program {
+    pub fn new() -> Self {
+        Self {
+            groups: Vec::new()
+        }
+    }
+}
+
+impl StatementBody {
+    pub fn new() -> Self {
+        Self {
+            blocks: Vec::new()
+        }
+    }
+}
+
+impl Block {
+    pub fn new() -> Self {
+        Self {
+            block_type: "".to_string(),
+            id: "".to_string(),
+            fields: None,
+            statements: None
+        }
+    }
+
+    pub fn add_field(&mut self, name: String, value: FieldValue) {
+        match self.fields {
+            Some(ref mut f) => {
+                f.insert(name.to_owned(), value);
```

Comment:
By removing `Option` from wrapping `fields` in the `Block` struct, this entire function's body just becomes the one line:

```
f.insert(name.to_owned(), value);
```